### PR TITLE
[DO NOT MERGE] serverless-webhooks

### PR DESCRIPTION
**Pull request by tjanczuk**
The serverless webhooks post in the Auth0 Extend category.

### CODE REVIEW

**Comment by tjanczuk** [comment]
image

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+### High HTTP fidelity
+
+Auth0 Extend offers very high fidelity with the HTTP protocol. The developer remains in full control of request and response headers and content type. If you can do it with webhooks and Node.js, you can also do it with Auth0 Extend - without additional translation layers. 
+
+![High HTTP fidelity](https://cloud.githubusercontent.com/assets/822369/26188037/a80c52ba-3b51-11e7-9d37-9d296fa2fc7f.png)
+
+### Secrets and configuration
+
+Majority of extensions of your plarform will communicate with external systems to do their job. It may be a Mongo database, Twilio or Sendgrid APIs, Salesforce, or any other system that expose endpoints. Auth0 Extend provides a very convenient and secure way of provisioning and handling API keys and other secret configuration necessary for extensions to do their job.
```
Comment (ggoodman) [inline _posts/2017-05-18-serverless-webhooks.markdown]: > *The* majority of...

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+![Secrets and configuration](https://cloud.githubusercontent.com/assets/822369/26188072/f918a9f6-3b51-11e7-8842-f2e3736e1de1.png)
+
+Secrets are encrypted at rest and stored alongside the code of the extension. They are only decrypted when the time comes to execute the extension, which provides security-in-depth for protecting your customers' data. 
+
+### Integrated runner
+
+Auth0 Extend editor includes a runner component that allows your customers to test their code without using external tools. When you integrate Auth0 Extend into your SaaS, you can configure the runner with sample data specific to your extensibility points, so that your users are only a click away from testing their code with correct payloads.
```
Comment (ggoodman) [inline _posts/2017-05-18-serverless-webhooks.markdown]: `s/correct/meaningful`?

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+### Real-time logging
+
+Testing of the code during development is further facilitated with access to real-time logging information generated by the code. This functionality is integrated into Auth0 Extend editor and can be used as needed by your users to capture any output generated by their code to *stdout*. 
+
+![Real-time logging](https://cloud.githubusercontent.com/assets/822369/26188164/86b545a8-3b52-11e7-813f-b31653d4df9d.png)
+
+### GitHub integration
+
+The Auth0 Extend system stores and manages the code your users write in order to execute it at runtime. But it also supports two-directional synchronization of that code with a GitHub repository.
```
Comment (ggoodman) [inline _posts/2017-05-18-serverless-webhooks.markdown]: `s/two-directional/bi-directional/`

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+Auth0 Extend middleware is a Swiss army knife that allows you to truly customize the programming experience you provide your users. 
+
+### White-labelled experience
+
+The Auth0 Extend editor itself is highly customizable and extensible. You can expose only the features you want to your users, and you can augment the look & feel of what remains. This includes using your own logo or CSS styles. 
+
+### Durable logs
+
+In addition to the real-time logs which provide your users with timely access to logging information during development, you can also configure Auth0 Extend to externalize all logging information to AWS Firehose. This provides you with the ability to store and process logs in Amazon S3, Redshift, or ElasticSearch for analytics and historical record.
```
Comment (ggoodman) [inline _posts/2017-05-18-serverless-webhooks.markdown]: `s/timely/immediate/`?

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+In addition to the real-time logs which provide your users with timely access to logging information during development, you can also configure Auth0 Extend to externalize all logging information to AWS Firehose. This provides you with the ability to store and process logs in Amazon S3, Redshift, or ElasticSearch for analytics and historical record. 
+
+### Battle tested technology
+
+Auth0 Extend is not a new product, it a result of scratching our own itch at Auth0. Auth0 Extend is based on the same [Webtask](https://webtask.io) technology we have developed and have been using at [Auth0](https://auth0.com) to power extensibility through custom code in our identity management platform. We've been running it for years (yes, it predates the word *serverless*), at scale, and around the world. We've taken it through the long and sometimes painful stabilization process so that we can offer the best in stability, reliability, failure recovery, security, and performance. 
+
+### Getting started
+
+Start integratig Auth0 Extend into your product today. We not only make your customer's life simple, we also care about your own experience integratig Auth0 Extend into your SaaS.
```
Comment (ggoodman) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Missing an `n` in integrating.

**Comment by landitus** [comment]
Looking at the blogpost I think there are two main issues that would drastically improve the post: 

1) The blogpost showcases the CRM app with old design >> we could mockup those images with the new design (once we decide on the color palette).

2) It also showcases the Editor, but the images get me a little bit confused: 
- The Editor is white and there's no clear divider between the image and the post. We should use the dark theme here. 
- The zoomed images make me loose track of where I am. I think we need the context of the widgets toolbar to in all images, or at least partially.

**Comment by vctrfrnndz** [comment]
IMO as this heavily features the Demo CRM app, I would have a way bigger impact if it goes out with the updated design and UX improvements.

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+The most popular mechanism used by SaaS products today to offer extensibility is based on webhooks. Webhooks enable the customer to create a web service that will be called when a certain event occurs in your platform. Webhooks are very powerful and flexible: they allow customers to implement arbitrary business logic to extend your core product. 
+
+![Today: webooks](https://cloud.githubusercontent.com/assets/822369/26187727/e9d40b7c-3b4f-11e7-8c3a-9e86b3560dc4.png)
+
+Yet webhooks have their disadvantages. Using webhooks requires your customers to invest in and address several layers of concerns related to running a web service: hosting, monitoring, securing, ensuring availability, and more. All this increases the barrier to extending your product and the required development and maintenance. 
+
+### Tomorrow: serverless webhooks with Auth0 Extend
+
+Auth0 Extend helps you add extensibility to your SaaS product in a way that addresses these concerns. Instead of having your customers provide a webhook URL to a service they need to host elsewhere, your users can implement their extension logic directly within your product using the Auth0 Extend editor.
```
Comment (sebadoom) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Link to Auth0 Extend docs or intro or landing.

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+While you can provide your own authoring experience for custom code and use only the secure, multi-tenant runtime Auth0 Extend provides to execute it, the Auth0 Extend editor is still the best way to highlight what the platform has to offer. Let's take a lap around Auth0 Extend from the perspective of your customers. 
+
+### Sky is the limit with Node.js and NPM
+
+The primary programming environment in Auth0 Extend is Node.js. Your users can use Node with the vast majority of Node.js modules on NPM across all the versions they desire. The Extend Editor provides a great authoring experience for JavaScript with all the usual creature comforts developers are used to, from syntax highlighting to intellisense. 
+
+![Sky is the limit with Node.js and NPM](https://cloud.githubusercontent.com/assets/822369/26187871/a193e05c-3b50-11e7-819a-b9b7ae417caf.png)
+
+Thanks to an innovative way of handling module dependencies by the Auth0 Extend runtime, the latency of code execution is much less affacted by the number or size of them compared to tranditional ways of provisioning Node.js applications.
```
Comment (sebadoom) [inline _posts/2017-05-18-serverless-webhooks.markdown]: affacted -> affected

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 1 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Thanks!

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 2 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: OK. thanks

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 3 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: thanks

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 4 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: thanks

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 5 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: thanks

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 6 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: thanks

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 7 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: thanks

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+> Add extensibility to your SaaS. Your users will thank you, your CFO will love you, and your boss will give you a raise.
+
+---
+
+### Today: webhooks
+
+The most popular mechanism used by SaaS products today to offer extensibility is based on webhooks. Webhooks enable the customer to create a web service that will be called when a certain event occurs in your platform. Webhooks are very powerful and flexible: they allow customers to implement arbitrary business logic to extend your core product. 
+
+![Today: webooks](https://cloud.githubusercontent.com/assets/822369/26187727/e9d40b7c-3b4f-11e7-8c3a-9e86b3560dc4.png)
```
Comment (stubailo) [inline _posts/2017-05-18-serverless-webhooks.markdown]: webhooks

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+### Tomorrow: serverless webhooks with Auth0 Extend
+
+Auth0 Extend helps you add extensibility to your SaaS product in a way that addresses these concerns. Instead of having your customers provide a webhook URL to a service they need to host elsewhere, your users can implement their extension logic directly within your product [using the Auth0 Extend editor](https://auth0.com/extend/try).
+
+![Tomorrow: serverless webhooks with Auth0 Extend](https://cloud.githubusercontent.com/assets/822369/26187776/2d57cc62-3b50-11e7-9e9b-12b88b474c84.png)
+
+Using Auth0 Extend is a win-win for you and your customers. Everybody benefits from the reduced time to market and the enablement of platform economy around your SaaS product. Your customers are going to be delighted with the state of the art, in-product development experience Auth0 Extend provides. They will appreciate being able to focus on business logic behind extending your system as opposed to servers, hosting, and maintence of their code. 
+
+### Use case: Apollo Lauchpad, the GraphQL playground
```
Comment (stubailo) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Launchpad

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+A great example of using Auth0 Extend to provide a sandboxed execution environment server side scripts is the just released [Apollo Launchpad](https://dev-blog.apollodata.com/introducing-launchpad-the-graphql-server-demo-platform-cc4e7481fcba), brought to you by the people behind [MeteorJS](https://www.meteor.com/).
```
Comment (stubailo) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Just "Meteor", not "MeteorJS"

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+Check out the [Apollo Launchpad](https://launchpad.graphql.com/new) yourself!
+
+### What exactly is Auth0 Extend? 
+
+Auth0 Extend is a managed service that can be integrated into your SaaS product to support authoring and execution of custom code your customers write to extend your platform. The service supports a rich development environment your customers can use to develop and debug Node.js code that you can embed directly in your web site. The second part of the Auth0 Extend service supports secure, "serverless" execution of these extensions on behalf of your customers, therefore removing all your customers' concerns related to hosting and maintaining a service. [Read more about the benefits of Auth0 Extend here](https://auth0.com/extend). 
+
+### Lap around Auth0 Extend
+
+While you can provide your own authoring experience for custom code and use only the secure, multi-tenant runtime Auth0 Extend provides to execute it, the Auth0 Extend editor is still the best way to highlight what the platform has to offer. Let's take a lap around Auth0 Extend from the perspective of your customers.
```
Comment (stubailo) [inline _posts/2017-05-18-serverless-webhooks.markdown]: > the Auth0 Extend editor is still the best way

to

> the Auth0 Extend editor is the best way

"still" is not helpful in this sentence, it cheapens the confusion.

File: _posts/2017-05-18-serverless-webhooks.markdown [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+
+### Lap around Auth0 Extend
+
+While you can provide your own authoring experience for custom code and use only the secure, multi-tenant runtime Auth0 Extend provides to execute it, the Auth0 Extend editor is still the best way to highlight what the platform has to offer. Let's take a lap around Auth0 Extend from the perspective of your customers. 
+
+### Sky is the limit with Node.js and NPM
+
+The primary programming environment in Auth0 Extend is Node.js. Your users can use Node with the vast majority of Node.js modules on NPM across all the versions they desire. The Extend Editor provides a great authoring experience for JavaScript with all the usual creature comforts developers are used to, from syntax highlighting to intellisense.
```
Comment (stubailo) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Not consistent about use of Node vs. Node.js

**Comment by stubailo** [comment]
All of my comments were marked outdated, but it's just cause the file was renamed.

**Comment by glennblock** [comment]
@stubailo great suggestion in terms of a deeper technical post in the future. For this post we wanted to focus on what Extend gives developers and point people to the new developer page.

**Comment by stubailo** [comment]
Oh sure, I'm saying my line-by-line comments are still relevant.

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 8 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: thanks!

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 9 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: Of course

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 10 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: ok

File: _posts/2017-05-18-serverless-webhooks.markdown (hunk 11 above)
Comment (tjanczuk) [inline _posts/2017-05-18-serverless-webhooks.markdown]: good point

**Pull request merged**